pontoon/sync/tasks.py

```diff
@@ -229,11 +224,18 @@ def sync_translations(
         repo_sync_log.end()
         return
 
+    # If project repositories have API access, we can retrieve latest commit hashes and detect
+    # changed locales before the expensive VCS pull/clone operations. When performing full scan,
+    # we still need to sync all locales.
+    if not full_scan:
+        locales = get_changed_locales(db_project, locales) or locales
```

Comment:
This will sync all locales if it should sync none.

This goes back to the changed_locales code returning None if any of the repos doesn't have an API. Maybe we should revert that logic and sync all locales for which we don't have an API return code, similar to what we're doing for http errors?

Then this `or locales` clause would work via get_changed_locales directly, and the "no locales need syncing" bug would be fixed.